docs/BootImageCAPIDesign.md

```
@@ -0,0 +1,239 @@
+    resource: machinedeployments
+    selection:
+      mode: All
+```
+
+---
+
+## CAPI Infrastructure Template Model
+
+CAPI does not embed provider specs inside `MachineSets` or `MachineDeployments`. Instead each resource holds an `infrastructureRef` pointing to a provider-specific template (e.g., `AWSMachineTemplate`, `AzureMachineTemplate`):
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)